### PR TITLE
Rebase onto Element iOS 1.8.12 tag

### PR DESCRIPTION
**Pull request by phlniji**
#496

### CODE REVIEW

File: Riot/Modules/Common/Recents/DataSources/RecentsDataSource.m [hunk 1]
```diff
@@ -727,7 +727,7 @@ - (UIView *)badgeViewForHeaderTitleInSection:(NSInteger)section
     if (numberOfNotifications)
     {
         UILabel *missedNotifAndUnreadBadgeLabel = [[UILabel alloc] init];
-        missedNotifAndUnreadBadgeLabel.textColor = ThemeService.shared.theme.baseTextPrimaryColor;
+        missedNotifAndUnreadBadgeLabel.textColor = ThemeService.shared.theme.tintContrastColor;
```
Comment (yostyle) [inline Riot/Modules/Common/Recents/DataSources/RecentsDataSource.m:730]: Why do we have this change in Tchap ?

File: Riot/Modules/SideMenu/SideMenuViewModel.swift [hunk 2]
```diff
@@ -107,6 +107,7 @@ final class SideMenuViewModel: SideMenuViewModelType {
         
         sideMenuItems += [
             .settings,
+            .help,
```
Comment (yostyle) [inline Riot/Modules/SideMenu/SideMenuViewModel.swift:110]: Do we need this item in Tchap ?

File: Riot/Modules/TabBar/TabBarCoordinator.swift [hunk 3]
```diff
@@ -701,8 +715,34 @@ final class TabBarCoordinator: NSObject, TabBarCoordinatorType {
...
+                    windowOverlay = WindowOverlayPresenter()
+                    let coachMarkView = CoachMarkView.instantiate(
+                        text: VectorL10n.sideMenuCoachMessage,
+                        from: CoachMarkView.TopLeftPosition,
+                        markPosition: .topLeft)
+                    windowOverlay?.show(coachMarkView, duration: 4.0)
+                }
+            }
+        }
+    }
```
Comment (yostyle) [inline Riot/Modules/TabBar/TabBarCoordinator.swift:745]: This method in never used in Tchap. you should probably comment it.

File: RiotSwiftUI/Modules/Spaces/SpaceCreation/SpaceCreationEmailInvites/Service/MatrixSDK/SpaceCreationEmailInvitesService.swift [hunk 4]
```diff
@@ -42,13 +42,14 @@ class SpaceCreationEmailInvitesService: SpaceCreationEmailInvitesServiceProtocol
...
-        isLoadingSubject.send(true)
-        session.prepareIdentityServiceForTerms(withDefault: RiotSettings.shared.identityServerUrlString) { [weak self] session, baseURL, accessToken in
-            self?.isLoadingSubject.send(false)
-            prepared?(baseURL, accessToken)
-        } failure: { [weak self] error in
-            self?.isLoadingSubject.send(false)
-            failure?(error)
-        }
+        // Tchap: Disable Spaces
+//        isLoadingSubject.send(true)
```
Comment (yostyle) [inline RiotSwiftUI/Modules/Spaces/SpaceCreation/SpaceCreationEmailInvites/Service/MatrixSDK/SpaceCreationEmailInvitesService.swift:46]: Could we ignore this file in project ?

File: RiotSwiftUI/Modules/Spaces/SpaceSettings/Coordinator/SpaceSettingsModalCoordinator.swift [hunk 5]
```diff
@@ -122,64 +122,69 @@ final class SpaceSettingsModalCoordinator: Coordinator {
     }
     
     private func exploreRooms(ofSpaceWithId spaceId: String) {
-        let coordinator = ExploreRoomCoordinator(session: parameters.session, spaceId: spaceId)
-        coordinator.delegate = self
-        add(childCoordinator: coordinator)
-        coordinator.start()
-        self.navigationRouter.present(coordinator.toPresentable(), animated: true)
+        // Tchap: Disable Spaces
```
Comment (yostyle) [inline RiotSwiftUI/Modules/Spaces/SpaceSettings/Coordinator/SpaceSettingsModalCoordinator.swift:125]: Could we ignore this file in project ?

File: Tchap/Modules/Home/HomeCoordinator.swift [hunk 6]
```diff
@@ -1,544 +1,544 @@
...
-    func globalSearchBar(_ globalSearchBar: GlobalSearchBar, textDidChange searchText: String?) {
-//        self.roomsCoordinator?.updateSearchText(searchText)
-        self.contactsCoordinator?.updateSearchText(searchText)
-    }
-}
-
-// MARK: - RoomsCoordinatorDelegate
-//extension HomeCoordinator: RoomsCoordinatorDelegate {
-//    func roomsCoordinator(_ coordinator: RoomsCoordinatorType, didSelectRoomID roomID: String) {
+///*
```
Comment (yostyle) [inline Tchap/Modules/Home/HomeCoordinator.swift:1]: Could we ignore this file in project instead of comment it ?

File: Tchap/Modules/Room/DataSources/RoomDataSource.m [hunk 7]
```diff
@@ -93,7 +93,7 @@ - (instancetype)initWithRoomId:(NSString *)roomId andMatrixSession:(MXSession *)
             
         }];
 
-//        [matrixSession.threadingService addDelegate:self];
+        [matrixSession.threadingService addDelegate:self];
```
Comment (yostyle) [inline Tchap/Modules/Room/DataSources/RoomDataSource.m:96]: Why do you use threadService here ?

File: Tchap/Modules/Room/DataSources/RoomDataSource.m [hunk 8]
```diff
@@ -110,7 +110,7 @@ - (void)finalizeInitialization
 
     // Sadly, we need to make sure we have fetched all room members from the HS
     // to be able to display read receipts
-    if (!self.isPeeking && ![self.mxSession.store hasLoadedAllRoomMembersForRoom:self.roomId])
+    if (![self.mxSession.store hasLoadedAllRoomMembersForRoom:self.roomId])
```
Comment (yostyle) [inline Tchap/Modules/Room/DataSources/RoomDataSource.m:113]: Why did you remove `isPeeking` check here ?

File: Tchap/Modules/Room/DataSources/RoomDataSource.m [hunk 9]
```diff
@@ -260,54 +239,48 @@ - (void)fetchEncryptionTrustedLevel
...
+        {
+            MXEvent *relatedEvent = [self.mxSession.store eventWithEventId:event.relatesTo.eventId
+                                                                    inRoom:event.roomId];
+            if (relatedEvent.isInThread)
+            {
+                //  ignore the event
+                return NO;
+            }
+        }
+    }
```
Comment (yostyle) [inline Tchap/Modules/Room/DataSources/RoomDataSource.m:283]: Why did you uncommented this code ?

File: Btchap/Config/BuildSettings.swift [hunk 10]
```diff
@@ -275,6 +314,10 @@ final class BuildSettings: NSObject {
     static var isRoomScreenEnableMessageBubblesByDefault: Bool {
         return self.roomScreenTimelineDefaultStyleIdentifier == .bubble
     }
+    static let roomScreenUseOnlyLatestUserAvatarAndName: Bool = false
+
+    /// Allow split view detail view stacking
+    static let allowSplitViewDetailsScreenStacking: Bool = true
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Btchap/Config/BuildSettings.swift:320]: Not at this time.

File: Btchap/Config/BuildSettings.swift [hunk 11]
```diff
@@ -324,20 +366,31 @@ final class BuildSettings: NSObject {
     static let authScreenShowPhoneNumber = true
     static let authScreenShowForgotPassword = true
     static let authScreenShowCustomServerOptions = true
+    static let authScreenShowSocialLoginSection = true
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Btchap/Config/BuildSettings.swift:369]: The authentication mechanism is completely different in Tchap than in Element. So, changing those values won't affect anything, but I agree to do this, because this might change in the future.

File: Btchap/Config/BuildSettings.swift [hunk 12]
```diff
@@ -275,6 +314,10 @@ final class BuildSettings: NSObject {
     static var isRoomScreenEnableMessageBubblesByDefault: Bool {
         return self.roomScreenTimelineDefaultStyleIdentifier == .bubble
     }
+    static let roomScreenUseOnlyLatestUserAvatarAndName: Bool = false
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Btchap/Config/BuildSettings.swift:317]: This feature has been done in Element for another fork to fix a case where avatar were not changed. A boolean has been added to have the same behavior than previously by default, but we can use it in Tchap if needed.

File: Riot/Generated/Vector_Strings.swift [hunk 13]
```diff
@@ -3856,9 +3924,9 @@ public class VectorL10n: NSObject {
     return VectorL10n.tr("Vector", "open") 
   }
   /// or
-//  public static var or: String {
-//    return VectorL10n.tr("Vector", "or")
-//  }
+  public static var or: String {
+    return VectorL10n.tr("Vector", "or")
+  }
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Riot/Generated/Vector_Strings.swift:3929]: This method previously causes build errors in some cases and were not used in Tchap. This is why it was commented before. But now, we no more have the same build issues, and it's required by used code added in Element in this rebase.

File: Riot/Managers/PushNotification/PushNotificationService.m [hunk 14]
```diff
@@ -486,27 +486,28 @@ - (void)handleNotificationInlineReplyForRoomId:(NSString*)roomId
...
+//                                                initialEventId:nil
+//                                                      threadId:threadId
+//                                              andMatrixSession:mxSession
+//                                                    onComplete:^(MXKRoomDataSource *threadDataSource) {
+//                    dataSource = threadDataSource;
+//                    dispatch_group_leave(dispatchGroupDataSource);
+//                }];
+//            }
+//            else
+//            {
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Riot/Managers/PushNotification/PushNotificationService.m:503]: To do this, we need to add the Thread files to the project. ThreadDataSource is not defined at this time without them.
We should probably discuss in the future about including files of non used features.

File: Tchap/Modules/Home/HomeCoordinator.swift (hunk 6 above)
Comment (phlniji) [reply Tchap/Modules/Home/HomeCoordinator.swift:1]: HomeCoordinator is not used anymore in the project, but still available in the repo. We should clean all the unused files when the re-alignment onto Element will be all finished (to avoid the lost of some Tchap code / use cases).

File: Tchap/Modules/Room/DataSources/RoomDataSource.m (hunk 7 above)
Comment (phlniji) [reply Tchap/Modules/Room/DataSources/RoomDataSource.m:96]: It's a mistake. I'll fix it

File: Tchap/Modules/Room/DataSources/RoomDataSource.m (hunk 8 above)
Comment (phlniji) [reply Tchap/Modules/Room/DataSources/RoomDataSource.m:113]: It's a mistake. I'll fix it

File: Tchap/Modules/Room/DataSources/RoomDataSource.m (hunk 9 above)
Comment (phlniji) [reply Tchap/Modules/Room/DataSources/RoomDataSource.m:283]: It's probably a merge issue. I'll revert this.

File: Btchap/Config/BuildSettings.swift (hunk 11 above)
[the remark this answers is not in the document]
Comment (yostyle) [reply Btchap/Config/BuildSettings.swift]: Yes at this time we don't use those parameters.

File: Btchap/Config/BuildSettings.swift (hunk 12 above)
[the remark this answers is not in the document]
Comment (yostyle) [reply Btchap/Config/BuildSettings.swift:317]: We may check this with @giomfo but it's a non-blocking issue at this time.

File: RiotSwiftUI/Modules/Spaces/SpaceSettings/Coordinator/SpaceSettingsModalCoordinator.swift (hunk 5 above)
Comment (yostyle) [reply RiotSwiftUI/Modules/Spaces/SpaceSettings/Coordinator/SpaceSettingsModalCoordinator.swift:125]: Please revert those changes.

File: Tchap/Modules/Home/HomeCoordinator.swift (hunk 6 above)
Comment (yostyle) [reply Tchap/Modules/Home/HomeCoordinator.swift:1]: You should restore this file and ignore it in the project

File: RiotSwiftUI/Modules/Spaces/SpaceCreation/SpaceCreationEmailInvites/Service/MatrixSDK/SpaceCreationEmailInvitesService.swift (hunk 4 above)
Comment (phlniji) [reply RiotSwiftUI/Modules/Spaces/SpaceCreation/SpaceCreationEmailInvites/Service/MatrixSDK/SpaceCreationEmailInvitesService.swift:46]: I cannot change this, because this change is mandatory to build the project, and this file cannot be removed from the target. There are a lots of dependencies with the Spaces feature everywhere inside the code.

File: RiotSwiftUI/Modules/Spaces/SpaceSettings/Coordinator/SpaceSettingsModalCoordinator.swift (hunk 5 above)
Comment (phlniji) [reply RiotSwiftUI/Modules/Spaces/SpaceSettings/Coordinator/SpaceSettingsModalCoordinator.swift:125]: I cannot change this, because this change is mandatory to build the project, and this file cannot be removed from the target. There are a lots of dependencies with the Spaces feature everywhere inside the code.

File: Riot/Modules/Application/LegacyAppDelegate.m [hunk 15]
```diff
@@ -2947,28 +3024,48 @@ - (void)showRoomWithParameters:(RoomNavigationParameters*)parameters completion:
...
+            MXWeakify(self);
+            [mxSession.matrixRestClient roomSummaryWith:roomId via:@[] success:^(MXPublicRoom *room) {
+                MXStrongifyAndReturnIfNil(self);
+                if ([room.roomTypeString isEqualToString:MXRoomTypeStringSpace])
+                {
+                    SpacePreviewNavigationParameters *spacePreviewNavigationParameters = [[SpacePreviewNavigationParameters alloc] initWithPublicRoom:room mxSession:mxSession senderId:parameters.senderId presentationParameters:parameters.presentationParameters];
+                    [self showSpacePreviewWithParameters:spacePreviewNavigationParameters];
+                }
+                else
+                {
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Riot/Modules/Application/LegacyAppDelegate.m]: Done in the latest commit

File: Riot/Modules/Room/RoomViewController.m [hunk 16]
```diff
@@ -7122,31 +7395,31 @@ - (void)roomInfoCoordinatorBridgePresenterDelegateDidLeaveRoom:(RoomInfoCoordina
...
+                    completion:(void (^)(BOOL))completion
+{
+    [self.roomDataSource sendVoiceMessage:url mimeType:nil duration:duration samples:samples success:^(NSString *eventId) {
+        MXLogDebug(@"Success with event id %@", eventId);
+        completion(YES);
+    } failure:^(NSError *error) {
+        MXLogError(@"Failed sending voice message");
+        completion(NO);
+    }];
+}
```
[the remark this answers is not in the document]
Comment (phlniji) [reply Riot/Modules/Room/RoomViewController.m]: Done too

**Review by yostyle — APPROVED** [review]
